xonsh/prompt/vc.py

```diff
@@ -52,23 +52,33 @@ def get_git_branch():
         branch = None
     return branch
 
+def _get_hg_root(q):
+    _curpwd = os.getcwd()
+    while True:
+        if any([b.name == '.hg' for b in os.scandir(_curpwd)]):
+            q.put(_curpwd)
+            break
+        else:
+            _oldpwd = _curpwd
+            _curpwd = os.path.split(_curpwd)[0]
+            if _oldpwd == _curpwd:
+                return False
```

Comment:
Ok. Looking at the code, it doesn't seem the queue thing makes a difference for the speed. I also recall this was changed originally due to speed issues. Can't recall who handled it. `git blame` can probably help us find the right issue.